python/ray/tune/trial.py

```diff
@@ -655,7 +656,7 @@ def _generate_dirname(self):
             generated_dirname = f"{str(self)}_{self.experiment_tag}"
             generated_dirname = generated_dirname[:MAX_LEN_IDENTIFIER]
             generated_dirname += f"_{date_str()}"
-        return generated_dirname.replace("/", "_")
+        return re.sub("[/()]", "_", generated_dirname)
```

Comment:
Can you add a comment here to describe the intention? I think it may be helpful in case there are other special character that cause problems with `rsync`, so maintainers will know to append characters here if they run into the same problem.

Reply:
Sounds good. Added :)